src/com/ms/silverking/io/FileUtil.java

```diff
@@ -21,216 +23,243 @@
+
+  public static void writeToFile(String fileName, String text) throws IOException {
+    writeToFile(new File(fileName), text);
+  }
+
+  public static void writeToFile(File file, String text) throws IOException {
+    writeToFile(file, text, false);
+  }
+
+  public static void writeToFile(File file, String text, boolean append) throws IOException {
+    try (BufferedWriter out = new BufferedWriter(new OutputStreamWriter(new FileOutputStream(file, append)))) {
+      out.write(text);
     }
-    
-    public static int readFileAsInt(File file) throws IOException {
-        return Integer.parseInt(readFileAsString(file).trim());
+  }
+
+  public static void writeToFile(String fileName, Collection lines) throws IOException {
+    writeToFile(new File(fileName), lines);
+  }
+
+  public static void writeToFile(File file, Collection lines) throws IOException {
+    try (BufferedWriter out = new BufferedWriter(new OutputStreamWriter(new FileOutputStream(file)))) {
+      for (Object line : lines) {
+        out.write(line.toString() + "\n");
+      }
     }
-    
-    public static String readFileAsString(String file) throws IOException {
-        return readFileAsString(new File(file));
+  }
+
+  public static File[] namesToFiles(String[] names) {
+    File[] files;
+
+    files = new File[names.length];
+    for (int i = 0; i < files.length; i++) {
+      files[i] = new File(names[i]);
     }
-    
-    public static String readFileAsString(File file) throws IOException {
-        return new String(readFileAsBytes(file));
+    return files;
+  }
+
+  public static byte[] readFileAsBytes(File file) throws IOException {
+    try (FileInputStream fIn = new FileInputStream(file)) {
+      byte[] buf;
+
+      buf = new byte[(int) file.length()];
+      StreamUtil.readBytes(buf, 0, buf.length, new BufferedInputStream(fIn));
+      return buf;
     }
+  }
+
+  public static int readFileAsInt(String file) throws IOException {
+    return readFileAsInt(new File(file));
+  }
+
+  public static int readFileAsInt(File file) throws IOException {
+    return Integer.parseInt(readFileAsString(file).trim());
+  }
+
+  public static String readFileAsString(String file) throws IOException {
+    return readFileAsString(new File(file));
+  }
+
+  public static String readFileAsString(File file) throws IOException {
+    return new String(readFileAsBytes(file));
+  }
+
+  public static List<String> readFileAsLineList(File file) throws IOException {
+    String s;
+    String[] lines;
 
-    public static List<String> readFileAsLineList(File file) throws IOException {
-        String    s;
-        String[]    lines;
-        
-        s = readFileAsString(file);
-        lines = s.split("\\r?\\n");
-        return ImmutableList.copyOf(lines);
+    s = readFileAsString(file);
+    lines = s.split("\\r?\\n");
+    return ImmutableList.copyOf(lines);
+  }
+
+  public static void cleanDirectory(File dir) {
+    for (File file : dir.listFiles()) {
+      file.delete();
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The result of deleting a file on this line should be checked and failure should be logged. java.io.File.delete() will simply return false if it fails to delete the file, for example, when the file does not exist. It will not throw unless there is a security exception.